dpnp/backend/kernels/dpnp_krnl_logic.cpp

```diff
@@ -29,53 +29,80 @@
 #include "dpnp_iface.hpp"
 #include "queue_sycl.hpp"
 
+
+template <typename _DataType, typename _ResultType>
+class dpnp_all_c_kernel;
+
 template <typename _DataType, typename _ResultType>
 void dpnp_all_c(void* array1_in, void* result1, const size_t size)
```

Comment:
```suggestion
void dpnp_all_c(const void* array1_in, void* result1, const size_t size)
```
Do the same in header file